Zend/zend_operators.c

```diff
@@ -1337,6 +1354,7 @@ ZEND_API int ZEND_FASTCALL bitwise_not_function(zval *result, zval *op1) /* {{{
 		default:
 			ZEND_TRY_UNARY_OBJECT_OPERATION(ZEND_BW_NOT);
 
+			ZVAL_UNDEF(result);
```

Comment:
Note that due to opcache optimizations `result == op1` is possible for `bitwise_not_function` as well. However, I don't think there is any case where this value being overwritten is observable. (Similarly, `op1` can't be a string in this case, which is why the code in the `IS_STRING` case is not problematic either.)